### PR TITLE
Save XML file only on XDMFFile.close()

### PR DESCRIPTION
**Pull request by michalhabera**
Avoids filesystem touching on every write operation.

### CODE REVIEW

**Comment by garth-wells** [comment]
I think we need a `flush` method that writes the pugi XML file without closing the `XDMFFile` - helpful if a computation get interrupted.

**Comment by michalhabera** [comment]
Ok, I'll add `XDMFFile.flush` that would flush both HDF and XML. The default behavior would then be to not flush, which is the performant way, but prone to corrupted files in case of interruption.

**Comment by michalhabera** [comment]
@garth-wells Flush added

**Comment by garth-wells** [comment]
I think we need to add `flush` to the Python interface too.

File: cpp/dolfinx/io/XDMFFile.cpp [hunk 1]
```diff
@@ -118,6 +118,8 @@ XDMFFile::~XDMFFile() { close(); }
 //-----------------------------------------------------------------------------
 void XDMFFile::close()
 {
+  flush();
```
Comment (garth-wells) [inline cpp/dolfinx/io/XDMFFile.cpp:121]: I don't think flush should be called here - no need to also flush the HDF5 file.

**Pull request merged**